benchmarks/harness/FileCheckParser/__init__.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+"""
+ Parse IR files for comments indicating metadata that we can actually use.
+
+ Usage:
+  params = FileCheckParser(self.logger).parse('kernel.mlir')
+
+ Params: A dictionary containing the following fields, if found
+    - 'iters': Iterations, from a RUN line
+    - 'entry': Kernel function name, from a RUN line
+    - 'libs': Shared library argument, from a RUN line
+    - 'expected_mean': From a BENCH_EXPECTED_MEAN line
+    - 'expected_stdev': From a BENCH_EXPECTED_STDEV line
+"""
+
+import re
+
+class FileCheckParser(object):
+    """Parsers IR files for FileCheck lines to extract informatio
+       about the execution of the kernel"""
+
+    def __init__(self, logger=None):
```

Comment:
nit: methods assume presence of a `logger` so default `None` might blow up

Reply:
good point, now it's compulsory